zerver/lib/validator.py

```diff
@@ -163,6 +163,24 @@ def f(var_name: str, val: object) -> List[ResultT]:
         return cast(List[ResultT], val)
     return f
 
+def check_mixed_list(sub_validators: List[Validator[ResultT]]) -> Validator[List[ResultT]]:
```

Comment:
I'm not sure `mixed` expresses the idea for this quite right; that sorta feels like the formats are all mixed up.  But I'm not sure I have something better either; maybe `check_tuple_typed_list`?

Reply:
Per chat going with check_list_with_sequence_of_types.